src/settings_parser/settings.cpp

```diff
@@ -0,0 +1,169 @@
+#include <fstream> // for file operations
+#include <iostream> // for cout
+#include <functional> // for hash
+#include "settings.h"
+
+void Settings::loadFromFile(std::string filename){
+    
+    // open file
+    std::ifstream file(filename.c_str(), std::ios::in);
+
+    // check if file is open
+    if (!file.is_open()){
+        std::cout << "Could not open parameter file \"" << filename << "\"." << std::endl;
+        return ;
+    }
+
+    // loop over lines of file
+    for (int lineNo = 0;; lineNo++){
+
+        // read line
+        std::string line;
+        getline(file, line);
+
+        // at the end of the file, break the "for" loop
+        if (file.eof())
+            break;
+
+        LineContent lineContent = Settings::readSingleLine(line);
+
+        if (!lineContent.isContent){
+            continue;
+        }
+
+        // parse actual value and set corresponding parameter
+        Settings::setParameter(lineContent.parameterName, lineContent.value);
+
+    }
+}
+
+
+void Settings::printSettings(){
+  std::cout << "Settings: " << std::endl
+    << "  physicalSize: " << physicalSize[0] << " x " << physicalSize[1] << ", nCells: " << nCells[0] << " x " << nCells[1] << std::endl
+    << "  endTime: " << endTime << " s, re: " << re << ", g: (" << g[0] << "," << g[1] << "), tau: " << tau << ", maximum dt: " << maximumDt << std::endl
+    << "  dirichletBC: bottom: (" << dirichletBcBottom[0] << "," << dirichletBcBottom[1]  << ")"
+    << ", top: ("  << dirichletBcTop[0] << "," << dirichletBcTop[1]  << ")"
+    << ", left: ("  << dirichletBcLeft[0] << "," << dirichletBcLeft[1] << ")"
+    << ", right: ("  << dirichletBcRight[0] << "," << dirichletBcRight[1] << ")" << std::endl
+    << "  useDonorCell: " << std::boolalpha << useDonorCell << ", alpha: " << alpha << std::endl
+    << "  pressureSolver: " << pressureSolver << ", omega: " << omega << ", epsilon: " << epsilon << ", maximumNumberOfIterations: " << maximumNumberOfIterations << std::endl;
+}
+
+
+Settings::LineContent Settings::readSingleLine(std::string line){
+    Settings::LineContent lineContent;
+    lineContent.isContent = false;
+
+    int idx_first_non_white = line.find_first_not_of(" \t"); // find first character that is not a whitespace
+    int idx_equalSign = line.find_first_of("=");
+
+    // Checks if line contains content
+
+    // if first character is a '#', skip line
+    if(line[0] == '#')
+        return lineContent;
+
+    // if line only contains whitespace, skip line
+    if(idx_first_non_white == std::string::npos) // line only contains whitespace
+        return lineContent;
+
+    // if line does not contain a '=' sign, skip line
+    if(idx_equalSign == std::string::npos)
+        return lineContent;
+
+
+    // --- Line contains content ---
+    // devide content on '=' and assign `parameterName` and `value` without whitespaces 
+    lineContent.isContent = true;
+
+
```

Comment:
I thought about first splitting on the `=`
```cpp
std::string rawName = std::substring(idx_first_non_white, idx_equal)
std::sting rawValue = ....
```
and then whip off the whitespaces and comments, but it bloated up everything and was ugly espescially because your approach is already good and fast.

I thought about
```cpp
lineContent.parameterName = line.substr(idx_first_non_white, line.find_first_of(" =\t"));
line.erase(0, line.find_first_of("=") + 1);
lineContent.value = line.substr(line.find_first_not_of(" \t"), line.find_first_of(" #\t"));
```
but then `find_first_of` might return wrong results, so this longer version is preferable (*and I don't want to use something else as `find_first_of` because that can be error heavy*)